Makefile

```diff
@@ -1,5 +1,10 @@
+# Enable `lto`.
+#
+# This is equivalent to `--profile prod`, but keeps the resulting binary in
+# `./target/release/neard`, to not break historical infra relying on that.
```

Comment:
Sounds like we can do `--prefile prod --target-dir target`? I remember I was looking into creating a prod profile a while back; don’t recall why we didn’t go with it then.  Another option would be to build in target/prod directory and then symlink executables to target/release

Reply:
Ah, no, that was you: https://github.com/near/nearcore/issues/6226

Reply:
And again, no, `--target-dir` won’t do.  It’s again issue with `--out-dir` being unstable.

Reply:
Yeah, I also think there's sadly no great solution here. The way I think about it, `--profile prod` is the way to build the right binary, and for `Makefile` we need to use _a_ work-around. Two work-arounds which would work: 

* `export` and use `-r`
* `--profile prod` and linking afterwards

I would say first-best solution is to put stuff in `./target/neard`, so that we are not dependent on Cargo's layout (eg, if we decide to cross-compile stuff today, we'd also observe different target layout). 

But this ideal world obviously won't fly as everyone is already depending on stuff being in `./target/release`.